frontend/talentsearch/src/js/components/experienceForm/ExperienceForm.tsx

```diff
@@ -0,0 +1,256 @@
+import React from "react";
+import { useIntl } from "react-intl";
+import { toast } from "react-toastify";
+import { SubmitHandler } from "react-hook-form";
+import isEqual from "lodash/isEqual";
+import { BasicForm, TextArea } from "@common/components/form";
+import { commonMessages } from "@common/messages";
+import { getLocale } from "@common/helpers/localize";
+import { navigate } from "@common/helpers/router";
+import useLocalStorage from "@common/hooks/useLocalStorage";
+
+import ProfileFormWrapper from "../applicantProfile/ProfileFormWrapper";
+import ProfileFormFooter from "../applicantProfile/ProfileFormFooter";
+
+import AwardDetailsForm from "../awardDetailsForm/AwardDetailsForm";
+import CommunityExperienceForm from "../communityExperienceForm/CommunityExperienceForm";
+import EducationExperienceForm from "../educationExperienceForm/EducationExperienceForm";
+import PersonalExperienceForm from "../personalExperienceForm/PersonalExperienceForm";
+import WorkExperienceForm from "../workExperienceForm/WorkExperienceForm";
+
+import ExperienceSkills from "./ExperienceSkills";
+import WatchFormValues from "./WatchFormValues";
+import type { Skill } from "../../api/generated";
+import {
+  useGetMyExperiencesQuery,
+  useGetMeQuery,
+  useGetSkillsQuery,
+} from "../../api/generated";
+import applicantProfileRoutes from "../../applicantProfileRoutes";
+
+import type {
+  ExperienceType,
+  AllFormValues,
+  FormValues,
+  ExperienceDetailsSubmissionData,
+  ExperienceMutationResponse,
+  ExperienceQueryData,
+  ExperienceDetailsDefaultValues,
+} from "./types";
+
+import queryResultToDefaultValues from "./defaultValues";
+import formValuesToSubmitData from "./submissionData";
+import useExperienceMutations from "./mutations";
+
+export interface ExperienceFormProps {
+  experienceType: ExperienceType;
+  experience?: ExperienceQueryData;
+  skills: Skill[];
+  onUpdateExperience: (values: ExperienceDetailsSubmissionData) => void;
+}
+
+export const ExperienceForm: React.FunctionComponent<ExperienceFormProps> = ({
+  experience,
+  experienceType,
+  onUpdateExperience,
+  skills,
+}) => {
+  const intl = useIntl();
+  const defaultValues = experience
+    ? queryResultToDefaultValues(experienceType, experience)
+    : { skills: undefined };
+  const [locallySavedForm, setLocallySavedForm] = useLocalStorage<
+    ExperienceDetailsDefaultValues | undefined
+  >(
+    "ts-createExperience", // unique storage key
+    defaultValues, // start form off empty
+  );
```

Comment:
Some thoughts about this form caching... 🤔 

It looks liked the cached values will override the defaultValues as long as the key exists in local storage. I can't test it right now, but it sounds like if I work on one Experience, and then I try to edit a second Experience, I'll be stuck looking at the previous one.

One way around this would be to use more storage keys. Maybe ``key = `ts-createExperience-${experience ? experience.id : experienceType}`;``

One way around this could be